test/strip-duplicate-spaces.spec.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        Tests for the stripDuplicateSpaces function
+ * CVM-Role:        TESTING
+ * Maintainers:     Rich Douglas
+ * License:         GNU GPL v3
+ *
+ * Description:     This file tests the stripDuplicateSpaces function.
+ *
+ * END HEADER
+ */
+
+import { EditorSelection, EditorState, Transaction } from '@codemirror/state'
+import { fail, deepEqual } from 'assert'
+import { stripDuplicateSpaces } from 'source/common/modules/markdown-editor/util/strip-duplicate-spaces'
+
+describe('MarkdownEditor#stripDuplicateSpaces()', function () {

```

Comment:
I've added unit tests for sunny day (positive) and rainy day (negative) test cases.

Simple enough to do so, if Hendrik generally approves the approach (perhaps with tweaks), then banging out the rest of the text transforms -- with tests to give us confidence -- will be straightforward.